### ISSUE TITLE
Change Request: `prefer-const` to not recommend const loop variable

### ISSUE THREAD

**Issue by stephencelis** [eslint/eslint#4333]
The `prefer-const` rule currently recommends that the following code use `const`:

``` javascript
for (let n of [1, 2, 3]) { console.log(n) }
```

This is misleading given the current state of how `const` works in Node (v5.0.0):

```
> for (let n of [1, 2, 3]) { console.log(n) }
1
2
3
undefined
> for (const n of [1, 2, 3]) { console.log(n) }
1
1
1
undefined
```

I recommend ignoring the rule for variables that are created in looping constructs.

(Affects ESLint v1.8.0)

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
Related https://github.com/eslint/eslint/issues/4088

**Comment by stephencelis** [comment]
Ah, I did a search but it must have been scoped to open tickets.

Maybe make this a configuration option?

**Comment by nzakas** [comment]
It depends on why you'd want such an option. This is a bug in Node.js (V8, really, in [nonstrict mode](https://github.com/nodejs/node/pull/3152#discussion_r41552213)), that will hopefully be getting fixed. It appears you can make it work correctly in Node.js by using strict mode. So, the fact this doesn't work in Node.js is a bug and `const` can be used here legally.

Are you suggesting the option to workaround the Node.js bug? If so, then I disagree with adding the option because this will eventually be fixed by them.

**Comment by stephencelis** [comment]
Ah, sorry, it was a matter of me skimming the thread before getting pulled away and being under the impression that the behavior depended on mode. If it's a bug that should be fixed in Node, no need to make this an option.
